adafruit_httpserver/response.py

```diff
@@ -64,21 +67,20 @@ def _construct_response_bytes(  # pylint: disable=too-many-arguments
         status: HTTPStatus = CommonHTTPStatus.OK_200,
         content_type: str = MIMEType.TYPE_TXT,
         content_length: Union[int, None] = None,
-        headers: Dict[str, str] = None,
+        headers: HTTPHeaders = None,
         body: str = "",
     ) -> bytes:
         """Constructs the response bytes from the given parameters."""
 
         response = f"{http_version} {status.code} {status.text}\r\n"
 
-        # Make a copy of the headers so that we don't modify the incoming dict
-        response_headers = {} if headers is None else headers.copy()
-
-        response_headers.setdefault("Content-Type", content_type)
-        response_headers.setdefault("Content-Length", content_length or len(body))
-        response_headers.setdefault("Connection", "close")
+        headers.setdefault("Content-Type", content_type)
+        headers.setdefault(
+            "Content-Length", content_length or len(body.encode("utf-8"))
```

Comment:
Can we avoid encoding the body twice?

Reply:
Nice spot, it is the little things that matter.
Thanks!

Changed in [this commit](https://github.com/adafruit/Adafruit_CircuitPython_HTTPServer/pull/29/commits/aeb95a293251192b4ab3cea61a8fb70d9cee4d51).